include/alpaka/kernel/Traits.hpp

```diff
@@ -236,145 +273,127 @@ namespace alpaka
-        std::cout << __func__ << " workDiv: " << workDiv << ", kernelFnObj: " << core::demangled<decltype(kernelFnObj)>
-                  << std::endl;
-#endif
-        return trait::CreateTaskKernel<TAcc, TWorkDiv, TKernelFnObj, TArgs...>::createTaskKernel(
+        KernelBundle<TKernelFn, TArgs...> const& kernelBundle
+            = alpaka::KernelBundle(kernelFnObj, std::forward(args)...);
+        // Return the task kernel
+        return trait::CreateTaskKernel<TAcc, TWorkDiv, KernelBundle<TKernelFn, TArgs...>>::createTaskKernel(
             workDiv,
-            kernelFnObj,
-            std::forward<TArgs>(args)...);
+            kernelBundle);
+    }
+
+    //! Executes the given kernel in the given queue.
+    //!
+    //! \tparam TAcc The accelerator type.
+    //! \tparam TQueue The queue type for work to be submitted to.
+    //! \tparam TWorkDiv The type of the work division.
+    //! \tparam TKernelFn The kernel object type, which includes the kernel function object and it's invocation
+    //! arguments.
+    //! \tparam TArgs The kernel invocation argument types pack.
+    //! \param queue The queue to enqueue the view copy task into.
+    //! \param workDiv The index domain work division.
+    //! \param kernelBundle The kernel object instance, which includes the kernel function object and it's
+    //! invocation arguments.
+    //!
+    template<typename TAcc, typename TQueue, typename TWorkDiv, typename TKernelFn, typename... TArgs>
+    ALPAKA_FN_HOST auto exec(
+        TQueue& queue,
+        TWorkDiv const& workDiv,
+        KernelBundle<TKernelFn, TArgs...> const& kernelBundle) -> void
+    {
+        enqueue(queue, createTaskKernel<TAcc>(workDiv, kernelBundle));
     }
 
 #if BOOST_COMP_CLANG
 #    pragma clang diagnostic push
 #    pragma clang diagnostic ignored                                                                                  \
         "-Wdocumentation" // clang does not support the syntax for variadic template arguments "args,..."
 #endif
-//! Executes the given kernel in the given queue.
-//!
+    //! Executes the given kernel in the given queue.
+    //!
 //! \tparam TAcc The accelerator type.
+//! \tparam TQueue The queue type for work to be submitted to.
+//! \tparam TWorkDiv The type of the work division.
+//! \tparam TKernelFn Kernel function object type.
+//! \tparam TArgs Kernel function object argument types as a parameter pack.
 //! \param queue The queue to enqueue the view copy task into.
 //! \param workDiv The index domain work division.
 //! \param kernelFnObj The kernel function object which should be executed.
 //! \param args,... The kernel invocation arguments.
 #if BOOST_COMP_CLANG
 #    pragma clang diagnostic pop
 #endif
-    template<typename TAcc, typename TQueue, typename TWorkDiv, typename TKernelFnObj, typename... TArgs>
-    ALPAKA_FN_HOST auto exec(TQueue& queue, TWorkDiv const& workDiv, TKernelFnObj const& kernelFnObj, TArgs&&... args)
```

Comment:
There should be no need to change `&&` to value type

Reply:
done, thanks.